### PR TITLE
Fix zeitwerk loader

### PR DESCRIPTION
**Pull request by hasghari**
Currently, none of the files in this gem are auto-loaded when this gem is included in the `Gemfile` as:

```ruby
gem 'cloudflare-rails`
```

The reason is that bundler will automatically require a file from the gem load path with exactly the same name, but the file `cloudflare-rails.rb` does not exist in this gem so none of the files are required. When I manually required the correct file using:

```ruby
gem 'cloudflare-rails`, require: 'cloudflare_rails'
```

I got the error `NameError: uninitialized constant CloudflareRails`

A quick look into Zeitwerk revealed that `loader.eager_load` should be called **after** the `CloudflareRails` constant has been defined.

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,12 +1,6 @@
-# copied from https://codingdaily.wordpress.com/2011/01/14/test-a-gem-with-the-rails-3-stack/
-$LOAD_PATH.unshift File.expand_path('../lib', __dir__)
```
Comment (hasghari) [inline spec/spec_helper.rb:2]: This is redundant as the gemspec already adds `lib` to the load path.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,12 +1,6 @@
...
-
 ENV['RAILS_ENV'] ||= 'test'
 
 require 'bundler/setup'
-Bundler.setup
```
Comment (hasghari) [inline spec/spec_helper.rb:7]: This is automatically called by the previous line.

File: lib/cloudflare_rails.rb [hunk 3]
```diff
@@ -1,7 +1,9 @@
 require 'zeitwerk'
```
Comment (modosc) [inline lib/cloudflare_rails.rb:1]: is there any advantage to just renaming this file to `cloudflare-rails.rb`?

Comment (hasghari) [reply lib/cloudflare_rails.rb:1]: That should be fine.

Comment (modosc) [reply lib/cloudflare_rails.rb:1]: i'll wait and try in a different pr, i'd like to get this out asap. thanks for your help!

**Comment by modosc** [comment]
@hasghari i just released `5.0.1` with your fix, thanks again!

**Pull request merged**